driver/driver.go

```diff
@@ -118,13 +126,19 @@ func FromSource(source models.Source) (Driver, error) {
 
 		s3Client := s3.NewFromConfig(cfg, s3Opts...)
 
+		var checksumAlgorithm types.ChecksumAlgorithm
+		if source.ChecksumAlgorithm != "" && !source.SkipS3Checksums {
+			checksumAlgorithm = types.ChecksumAlgorithm(source.ChecksumAlgorithm)
+		}
```

Comment:
We should check if the resulting `checksumAlgorithm` matches one of the `types.ChecksumAlgorithm`. As-is, someone can pass in anything and we'll happily pass it along.